main.tf

```
@@ -1,54 +1,30 @@
+provider "azurerm" {
+  version = "1.22.1"
+}
+
 locals {
   s2sUrl = "http://rpe-service-auth-provider-${var.env}.service.core-compute-${var.env}.internal"
   # list of the thumbprints of the SSL certificates that should be accepted by the API (gateway)
-  thumbprints_in_quotes = "${formatlist("&quot;%s&quot;", var.api_gateway_test_certificate_thumbprints)}"
-  thumbprints_in_quotes_str = "${join(",", local.thumbprints_in_quotes)}"
-  api_policy = "${replace(file("template/api-policy.xml"), "ALLOWED_CERTIFICATE_THUMBPRINTS", local.thumbprints_in_quotes_str)}"
-  api_base_path = "payments-api"
-  dummy = "dummy"
+  payments_thumbprints_in_quotes     = formatlist("&quot;%s&quot;", var.payments_test_certificate_thumbprints)
+  payments_thumbprints_in_quotes_str = join(",", local.payments_thumbprints_in_quotes)
+
+  telephony_thumbprints_in_quotes     = formatlist("&quot;%s&quot;", var.telephony_api_gateway_certificate_thumbprints)
+  telephony_thumbprints_in_quotes_str = join(",", local.telephony_thumbprints_in_quotes)
+
+  bulkscanning_thumbprints_in_quotes     = formatlist("&quot;%s&quot;", var.bulkscanning_api_gateway_certificate_thumbprints)
+  bulkscanning_thumbprints_in_quotes_str = join(",", local.bulkscanning_thumbprints_in_quotes)
 }
 data "azurerm_key_vault" "payment_key_vault" {
-  name = "ccpay-${var.env}"
+  name                = "ccpay-${var.env}"
   resource_group_name = "ccpay-${var.env}"
 }
 
 data "azurerm_key_vault_secret" "s2s_client_secret" {
-  name = "gateway-s2s-client-secret"
-  vault_uri = "${data.azurerm_key_vault.payment_key_vault.vault_uri}"
+  name         = "gateway-s2s-client-secret"
+  key_vault_id = data.azurerm_key_vault.payment_key_vault.id
```

the terraform version on jenkins doesn't support this (needs upgrading)
```suggestion
  key_vault_id = "${data.azurerm_key_vault.payment_key_vault.id}"
```